src/modules/facility-covenant/dto/get-facility-covenants-response.dto.ts

```diff
@@ -0,0 +1,69 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { ENUMS } from '@ukef/constants';
+import { DateOnlyString } from '@ukef/helpers/date-only-string.type';
+import { IsEnum } from 'class-validator';
+
+export class GetFacilityCovenantsResponseDto {
+  @ApiProperty({
+    readOnly: true,
+    description:
+      'The identifier of the covenant in ACBS. When creating a covenant an identifier from the number generator should be used (if this value is skipped it will be auto-generated by ACBS without checking if it is already used).',
+  })
+  covenantIdentifier: string;
+
+  @ApiProperty({
+    readOnly: true,
+    description:
+      'The covenant type code: 43 for a UK Contract Value covenant, 46 for a Chargable Amount covenant, and 47 for a chargable Amount covenant not in GBP.',
+  })
+  @IsEnum(ENUMS.COVENANT_TYPE_CODES)
+  covenantType: string;
+
+  @ApiProperty({ readOnly: true, description: 'The identifier of the facility in ACBS. This will be a 10-digit code.' })
+  facilityIdentifier: string;
+
+  @ApiProperty({
+    readOnly: true,
+    description: 'The identifier of the portfolio in ACBS. The UKEF portfolio is E1. The maximum number of characters allowed is 2.',
+  })
+  portfolioIdentifier: string;
+
+  @ApiProperty({
+    readOnly: true,
+    description: 'The target amount used to determine if the covenant is in compliance or not. It is called target amount in ACBS.',
+  })
+  maximumLiability: number;
+
+  @ApiProperty({
+    readOnly: true,
+    description: 'The covenant currency type code. The maximum number of characters allowed is 3. It is called pledge type code in ACBS.',
+  })
+  currency: string;
+
+  @ApiProperty({
+    readOnly: true,
+    description: 'The effective date of the covenant. It is called effective date in ACBS. Note that guaranteeCommencementDate and effectiveDate are the same.',
+    type: Date,
+    nullable: true,
+    format: 'date',
+  })
+  guaranteeCommencementDate: DateOnlyString | null;
```

Comment:
For GETs, should we assume that all dates could possibly be null? Personally I don't see any disadvantages in doing so.

Reply:
Raised question on slack as it was also on another PR. I don't mind either way, just wanted to give more background details.

Reply:
As discussed on Slack, for a given field, we're going to do a null check if and only if Mulesoft does a null check for that field. So I've removed the null checks and tests for the date fields for this endpoint, as the Mulesoft code assumes that they aren't null.